src/data/real/dedekind_cut.lean

```diff
@@ -0,0 +1,403 @@
+/-
+Copyright (c) 2022 Yakov Pechersky. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yakov Pechersky
+-/
+import algebra.order.archimedean
+import data.set.pointwise
+import order.upper_lower
+
+/-!
+# Dedeking cuts
+
+A basic theory of Dedekind cuts, used in a construction of the reals.
+
+## Important definitions
+
+- `dedekind_cut Q` is an `add_comm_group` when `Q` is `linear_ordered_field` and `archimedean`
+- TODO: finish constructing the full field
+
+## Tags
+
+dedekind, real, cut
+
+## References
+
+[Rudin, *Principles of Mathematical Analysis* (Chapter 1, Appendix)][rudin1976]
+
+-/
+
+section
+
+variables (Q : Type*) [has_lt Q]
```

Comment:
I think `Q` isn't a very good name for a type. Mathlib convention says you should use something like `α` instead. It's extra confusing here, since Dedekind cuts are most often defined for `ℚ`, but that's not what you're doing here.

Reply:
https://leanprover-community.github.io/contribute/naming.html#variable-conventions
In particular, the latter portion regarding `G`, `R`, etc. I will rename to `P` for `poset`.

Reply:
I don't think partial orders are specific enough to count as anything other than a "generic type". Many if not most of the order files just use Greek letters for their types.

Reply:
As that link indicates, using greek letters is a possibility, not the only possibility. The reference I have listed uses greek letters for set names. _Introduction to Lattices and Order_ uses P for posets. If I rename the carrier type to alpha, which letters would be acceptable to use for terms of type `set alpha`? How about for terms of type `alpha`?

Reply:
We try to adhere to mathlib convention, rather than the conventions of any references, unless those are overwhelmingly common – and even then, we sometimes make exceptions (e.g. ordinals use `a`, `b`, `c`, instead of Greek letters). 

You can always use `a b c x y z : α` and `s t : set α`, which I believe is what most of mathlib does.